tests/performance_tests/performance_test.py

```diff
@@ -11,203 +12,463 @@
+                    break
+            lastMaxTpsAchieved = binSearchResults.maxTpsAchieved
+
+        analysisFinish = datetime.utcnow()
+
+        def calcLocalMax(threadToMaxDict: dict):
+            localMax = 0
+            recThreadCount = 0
+            for threads, tps in threadToMaxDict.items():
+                if tps > localMax:
+                    localMax = tps
+                    recThreadCount = threads
+                else:
+                    break
+            return recThreadCount
+
+        recommendedThreadCount = calcLocalMax(threadToMaxDict=threadToMaxTpsDict)
+
+        return PerformanceTest.PluginThreadOptResult(recommendedThreadCount=recommendedThreadCount, threadToMaxTpsDict=threadToMaxTpsDict,
+                                                     analysisStart=analysisStart, analysisFinish=analysisFinish)
+
+    def createTpsTestReport(self, tpsTestResult: TpsTestResult) -> dict:
+        report = {}
+        report['InitialMaxTpsAchieved'] = tpsTestResult.binSearchResults.maxTpsAchieved
+        report['LongRunningMaxTpsAchieved'] = tpsTestResult.longRunningSearchResults.maxTpsAchieved
+        report['tpsTestStart'] = tpsTestResult.tpsTestStart
+        report['tpsTestFinish'] = tpsTestResult.tpsTestFinish
+        report['InitialSearchResults'] =  {x: asdict(tpsTestResult.binSearchResults.searchResults[x]) for x in range(len(tpsTestResult.binSearchResults.searchResults))}
+        report['InitialMaxTpsReport'] =  tpsTestResult.binSearchResults.maxTpsReport
+        report['LongRunningSearchResults'] =  {x: asdict(tpsTestResult.longRunningSearchResults.searchResults[x]) for x in range(len(tpsTestResult.longRunningSearchResults.searchResults))}
+        report['LongRunningMaxTpsReport'] =  tpsTestResult.longRunningSearchResults.maxTpsReport
+        return report
+
+    def createReport(self,producerThreadResult: PluginThreadOptResult=None, chainThreadResult: PluginThreadOptResult=None, netThreadResult: PluginThreadOptResult=None, tpsTestResult: dict=None) -> dict:
+        report = {}
+        report['perfTestsBegin'] = self.testsStart
+        report['perfTestsFinish'] = self.testsFinish
+        if tpsTestResult is not None:
+            report.update(self.createTpsTestReport(tpsTestResult))
+
+        if producerThreadResult is not None:
+            report['ProducerThreadAnalysis'] = asdict(producerThreadResult)
+
+        if chainThreadResult is not None:
+            report['ChainThreadAnalysis'] = asdict(chainThreadResult)
+
+        if netThreadResult is not None:
+            report['NetThreadAnalysis'] = asdict(netThreadResult)
+
+        report['args'] =  self.prepArgsDict()
+        report['env'] = {'system': system(), 'os': os.name, 'release': release(), 'logical_cpu_count': os.cpu_count()}
+        report['nodeosVersion'] = Utils.getNodeosVersion()
+        return report
+
+    def reportAsJSON(self, report: dict) -> json:
+        if 'ProducerThreadAnalysis' in report:
```

Comment:
Rather than all of this, create a local datetime encoder object inheriting from `json.JSONEncoder` and implement its `default()` method to call `isoformat()` if the object type matches, then pass the class name to `json.dumps` as the `cls` argument.

Reply:
Nice. Yes, I like that. Thanks.
Addressed: https://github.com/AntelopeIO/leap/pull/510/commits/d291e17022589485a321c1e6d4b7cefde3a6fd1b